dtm0/recovery.c

```diff
@@ -0,0 +1,2170 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+/**
+   @page dtm0br-dld DLD of DTM0 basic recovery
+
+   - @ref DTM0BR-ovw
+   - @ref DTM0BR-def
+   - @ref DTM0BR-req
+   - @ref DTM0BR-depends
+   - @ref DTM0BR-highlights
+   - @subpage DTM0BR-fspec "Functional Specification"
+   - @ref DTM0BR-lspec
+      - @ref DTM0BR-lspec-comps
+      - @ref DTM0BR-lspec-rem-fom
+      - @ref DTM0BR-lspec-loc-fom
+      - @ref DTM0BR-lspec-evc-fom
+      - @ref DTM0BR-lspec-state
+      - @ref DTM0BR-lspec-thread
+      - @ref DTM0BR-lspec-numa
+   - @ref DTM0BR-conformance
+   - @ref DTM0BR-usecases
+   - @ref DTM0BR-ref
+   - @ref DTM0BR-impl-plan
+--->>>
+Max: [* defect *] ut and st sections are missing.
+IvanA: Agree. I will add it once we are more-or-less confident in use-cases and
+       requirements.
+<<<---
+--->>>
+Max: [* style *] usecases section should go to .h file, along with the rest of
+     the sections in .h in DLD template.
+<<<---
```

Comment:
I am not sure if these discussion should be removed before the final merge.

Reply:
Need to discuss this with Nikita and Max.

Reply:
We think this should be left as is and reworked in further pull requests.